reps/2023-03-10-batch-inference.md

```diff
@@ -0,0 +1,246 @@
+# Revamping APIs for Offline Batch Inference
+
+## Summary
+Reduce the friction for batch inference of pre-trained by updating the APIs to do the following:
+1. Remove AIR Checkpoints from the critical path for batch prediction.
+2. Simpler interface for chaining multiple preprocessing and prediction stages.
+
+### General Motivation
+There are changes to Ray Datasets in 2.3/2.4:
+1. Ray Datasets became lazy by default in Ray 2.3. Execution is no longer triggered until consumption is explicitly called.
+2. In Ray 2.4, Ray Datasets will move to streaming execution by default, resulting in there no longer being a separation between `Dataset` and `DatasetPipeline`. 
+3. In Ray 2.4, Ray Datasets will support a `"zero_copy"` batch format, allowing the core Ray Data to decide the most optimal batch format to use.
```

Comment:
We don't show how this is used in the REP--- should we either remove this line, or incorporate it into the REP?

Reply:
removed